apps/src/dance/ai/DanceAiClient.js

```diff
@@ -10,93 +10,78 @@ import CachedForegrounds from '@cdo/static/dance/ai/model/cached-spacy-foregroun
  * example: {"backgroundEffect":"sparkles","backgroundColor":"cool","foregroundEffect":"bubbles"}
  */
 export function chooseEffects(emojis) {
-  var foregroundValues = {};
-  var backgroundValues = {};
-  var paletteValues = {};
-  reformatInputs(foregroundValues, CachedForegrounds);
-  reformatInputs(backgroundValues, CachedBackgrounds);
-  reformatInputs(paletteValues, CachedPalettes);
+  // Obtain final summed output weight based off input received
+  const outputTypes = [CachedBackgrounds, CachedForegrounds, CachedPalettes];
+  const outputWeights = [];
+  outputTypes.forEach(outputJson => {
+    outputWeights.push(calculateOutputWeightsVector(emojis, outputJson));
+  });
 
-  const bestBackground = chooseHighestSum(backgroundValues, emojis);
-  const bestForeground = chooseHighestSum(foregroundValues, emojis);
-  const bestPalette = chooseHighestSum(paletteValues, emojis);
+  // Sort and slice top scoring options, mapped to their output identifiers (e.g. [[0.25, 'squiggles'], ...])
+  const numRandomOptions = 3;
+  const outputOptions = [];
+  outputWeights.forEach((weightVector, i) => {
+    outputOptions.push(
+      obtainTopOptions(numRandomOptions, weightVector, outputTypes[i])
+    );
+  });
+
+  // Clarification of which array correlates to which option
+  const backgroundOptions = outputOptions[0];
+  const foregroundOptions = outputOptions[1];
+  const paletteOptions = outputOptions[2];
 
+  // Choose random value from top scoring options
   const chosenEffects = {
-    backgroundEffect: bestBackground.title,
-    backgroundColor: bestPalette.title,
-    foregroundEffect: bestForeground.title,
+    backgroundEffect:
+      backgroundOptions[
+        Math.floor(Math.random() * backgroundOptions.length)
+      ][1],
```

Comment:
Looks like we're just reading back the name of the randomly chosen effect; do we need to return the output weight at all in that case? Can `obtainTopOptions` just return a list of effect names?

Reply:
Originally, I kept that output weight in for debugging purposes BUT it will also be helpful to reference that numeric weight when it comes time to generate explanation pages/graphs

Reply:
Going to hold out on removing numeric weights from the tuple for now as Sam mentioned there is a possibility that we might dynamically adjust the number of provided options based off a TBD threshold (e.g. if two options are weighed at 1.5 but the 3rd option is .25, drop to only 3. Meanwhile, if 5 options are all relatively high, possibly provide all five)